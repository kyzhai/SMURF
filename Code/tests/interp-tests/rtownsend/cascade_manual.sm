track1 = [[(0,2)$8],[(2,2)$8],[(4,2)$8],[(5,2)$8],[(7,2)$8],[(9,2)$8],[(11,2)$8], \
				  [(0,3)$(1 $+ 8)],[(-1,0)$4],[(0,3)$4],[(-1,0)$4],[(11,2)$4],[(-1,0)$4],[(11,2)$2]]

track2 = [[(-1,0)$1],[(11,2)$8],[(9,2)$(2..)],[(-1,0)$4 ],[(8,2)$4], [(-1,0)$4],[(8,2)$4], \
				  [(-1,0)$4],[(7,2)$2]]

track3 = [[(-1,0)$1],[(-1,0)$4],[(7,2)$8],[(5,2)$(2 $+ 8)],[(-1,0)$4 ],[(5,2)$4], [(-1,0)$4],[(5,2)$4], \
          [(-1,0)$4],[(4,2)$2]]

track4 = [[(-1,0)$1],[(-1,0)$2],[(4,2)$8],[(2,2)$4.],[(-1,0)$4 ],[(2,2)$4], [(-1,0)$4],[(2,2)$4], \
          [(-1,0)$4],[(0,2)$2]]

main = [track1,track2,track3,track4]
