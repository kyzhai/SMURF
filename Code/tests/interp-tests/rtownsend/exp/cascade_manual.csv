***** Generated by SMURF *****
number of trace: 4
Time Resolution (pulses per quarter note),4,
track 1,40,track 2,40,track 3,40,track 4,40,
Tick, Note (0-127), Velocity (0-127), Tick, Note (0-127), Velocity (0-127), Tick, Note (0-127), Velocity (0-127), Tick, Note (0-127), Velocity (0-127), 
0,60,90,,,,,,,,,,
2,60,0,,,,,,,,,,
2,62,90,,,,,,,,,,
4,62,0,,,,,,,,,,
4,64,90,,,,,,,,,,
6,64,0,,,,,,,,,,
6,65,90,,,,,,,,,,
8,65,0,,,,,,,,,,
8,67,90,,,,,,,,,,
10,67,0,,,,,,,,,,
10,69,90,,,,,,,,,,
12,69,0,,,,,,,,,,
12,71,90,,,,,,,,,,
14,71,0,,,,,,,,,,
14,72,90,,,,,,,,,,
30,72,0,,,,,,,,,,
,,,16,71,90,,,,,,,
,,,18,71,0,,,,,,,
,,,18,69,90,,,,,,,
,,,30,69,0,,,,,,,
,,,,,,20,67,90,,,,
,,,,,,22,67,0,,,,
,,,,,,22,65,90,,,,
,,,,,,30,65,0,,,,
,,,,,,,,,24,64,90,
,,,,,,,,,26,64,0,
,,,,,,,,,26,62,90,
,,,,,,,,,30,62,0,
34,72,90,34,68,90,34,65,90,34,62,90,
38,72,0,38,68,0,38,65,0,38,62,0,
42,71,90,42,68,90,42,65,90,42,62,90,
46,71,0,46,68,0,46,65,0,46,62,0,
50,71,90,50,67,90,50,64,90,50,60,90,
54,71,0,54,67,0,54,64,0,54,60,0,

