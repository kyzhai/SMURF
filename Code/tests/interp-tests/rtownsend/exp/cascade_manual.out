
Timing Resolution set to 4 PPQ

Instrument set to 48 on channel 0
Instrument set to 48 on channel 1
Instrument set to 48 on channel 2
Instrument set to 48 on channel 3

Track 0:
 tick 0, channel 1: program change 48
 tick 0, channel 1: note C4 on velocity: 90
 tick 2, channel 1: note C4 on velocity: 0
 tick 2, channel 1: note D4 on velocity: 90
 tick 4, channel 1: note D4 on velocity: 0
 tick 4, channel 1: note E4 on velocity: 90
 tick 6, channel 1: note E4 on velocity: 0
 tick 6, channel 1: note F4 on velocity: 90
 tick 8, channel 1: note F4 on velocity: 0
 tick 8, channel 1: note G4 on velocity: 90
 tick 10, channel 1: note G4 on velocity: 0
 tick 10, channel 1: note A4 on velocity: 90
 tick 12, channel 1: note A4 on velocity: 0
 tick 12, channel 1: note B4 on velocity: 90
 tick 14, channel 1: note B4 on velocity: 0
 tick 14, channel 1: note C5 on velocity: 90
 tick 32, channel 1: note C5 on velocity: 0
 tick 36, channel 1: note C5 on velocity: 90
 tick 40, channel 1: note C5 on velocity: 0
 tick 44, channel 1: note B4 on velocity: 90
 tick 48, channel 1: note B4 on velocity: 0
 tick 52, channel 1: note B4 on velocity: 90
 tick 60, channel 1: note B4 on velocity: 0
 tick 60, end of track
Track 1:
 tick 0, channel 2: program change 48
 tick 16, channel 2: note B4 on velocity: 90
 tick 18, channel 2: note B4 on velocity: 0
 tick 18, channel 2: note A4 on velocity: 90
 tick 32, channel 2: note A4 on velocity: 0
 tick 36, channel 2: note G#4 on velocity: 90
 tick 40, channel 2: note G#4 on velocity: 0
 tick 44, channel 2: note G#4 on velocity: 90
 tick 48, channel 2: note G#4 on velocity: 0
 tick 52, channel 2: note G4 on velocity: 90
 tick 60, channel 2: note G4 on velocity: 0
 tick 60, end of track
Track 2:
 tick 0, channel 3: program change 48
 tick 20, channel 3: note G4 on velocity: 90
 tick 22, channel 3: note G4 on velocity: 0
 tick 22, channel 3: note F4 on velocity: 90
 tick 32, channel 3: note F4 on velocity: 0
 tick 36, channel 3: note F4 on velocity: 90
 tick 40, channel 3: note F4 on velocity: 0
 tick 44, channel 3: note F4 on velocity: 90
 tick 48, channel 3: note F4 on velocity: 0
 tick 52, channel 3: note E4 on velocity: 90
 tick 60, channel 3: note E4 on velocity: 0
 tick 60, end of track
Track 3:
 tick 0, channel 4: program change 48
 tick 24, channel 4: note E4 on velocity: 90
 tick 26, channel 4: note E4 on velocity: 0
 tick 26, channel 4: note D4 on velocity: 90
 tick 32, channel 4: note D4 on velocity: 0
 tick 36, channel 4: note D4 on velocity: 90
 tick 40, channel 4: note D4 on velocity: 0
 tick 44, channel 4: note D4 on velocity: 90
 tick 48, channel 4: note D4 on velocity: 0
 tick 52, channel 4: note C4 on velocity: 90
 tick 60, channel 4: note C4 on velocity: 0
 tick 60, end of track
Program: 
	SFuncdec: 
		makeNotes ([]) (_) (_) :: [Int] -> [Int] -> [Beat] -> [Note] = []
	New Scope: 
		
	
	SFuncdec: 
		makeNotes (_) ([]) (_) :: [Int] -> [Int] -> [Beat] -> [Note] = []
	New Scope: 
		
	
	SFuncdec: 
		makeNotes (_) (_) ([]) :: [Int] -> [Int] -> [Beat] -> [Note] = []
	New Scope: 
		
	
	SFuncdec: 
		makeNotes (pc : pcs) (reg : regs) (bt : bts) :: [Int] -> [Int] -> [Beat] -> [Note] =  (pc, reg)$bt : makeNotes bts regs pcs
	New Scope: 
		ID: pc  :: Int
		ID: pcs  :: [Int]
		ID: reg  :: Int
		ID: regs  :: [Int]
		ID: bt  :: Beat
		ID: bts  :: [Beat]

	
	SFuncdec: 
		tail (hd : tl) :: [a] -> [a] = tl
	New Scope: 
		ID: hd  :: a
		ID: tl  :: [a]

	
	SFuncdec: 
		head (hd : tl) :: [a] -> a = hd
	New Scope: 
		ID: hd  :: a
		ID: tl  :: [a]

	
	SFuncdec: 
		reverse ([]) :: [a] -> [a] = []
	New Scope: 
		
	
	SFuncdec: 
		reverse (a : rest) :: [a] -> [a] = reverse rest ++ [a]
	New Scope: 
		ID: a  :: a
		ID: rest  :: [a]

	
	SFuncdec: 
		drop (0) (x) :: Int -> [Note] -> [Note] = x
	New Scope: 
		ID: x  :: [Note]

	
	SFuncdec: 
		drop (_) ([]) :: Int -> [Note] -> [Note] = []
	New Scope: 
		
	
	SFuncdec: 
		drop (x) (l : rest) :: Int -> [Note] -> [Note] = drop rest (x - 1)
	New Scope: 
		ID: x  :: Int
		ID: l  :: Note
		ID: rest  :: [Note]

	
	SFuncdec: 
		take (0) (_) :: Int -> [Note] -> [Note] = []
	New Scope: 
		
	
	SFuncdec: 
		take (_) ([]) :: Int -> [Note] -> [Note] = []
	New Scope: 
		
	
	SFuncdec: 
		take (x) (l : rest) :: Int -> [Note] -> [Note] = l : take rest (x - 1)
	New Scope: 
		ID: x  :: Int
		ID: l  :: Note
		ID: rest  :: [Note]

	
	SVardef: 
		ID: track1  :: System

	[[ (0, 2)$8], [ (2, 2)$8], [ (4, 2)$8], [ (5, 2)$8], [ (7, 2)$8], [ (9, 2)$8], [ (11, 2)$8], [ (0, 3)$1 $+ 8], [ (-1, 0)$4], [ (0, 3)$4], [ (-1, 0)$4], [ (11, 2)$4], [ (-1, 0)$4], [ (11, 2)$2]]
	SVardef: 
		ID: track2  :: System

	[[ (-1, 0)$1], [ (11, 2)$8], [ (9, 2)$2..], [ (-1, 0)$4], [ (8, 2)$4], [ (-1, 0)$4], [ (8, 2)$4], [ (-1, 0)$4], [ (7, 2)$2]]
	SVardef: 
		ID: track3  :: System

	[[ (-1, 0)$1], [ (-1, 0)$4], [ (7, 2)$8], [ (5, 2)$2 $+ 8], [ (-1, 0)$4], [ (5, 2)$4], [ (-1, 0)$4], [ (5, 2)$4], [ (-1, 0)$4], [ (4, 2)$2]]
	SVardef: 
		ID: track4  :: System

	[[ (-1, 0)$1], [ (-1, 0)$2], [ (4, 2)$8], [ (2, 2)$4.], [ (-1, 0)$4], [ (2, 2)$4], [ (-1, 0)$4], [ (2, 2)$4], [ (-1, 0)$4], [ (0, 2)$2]]
	SMain: [track1, track2, track3, track4]
Global Scope: 
	ID: main  :: [System]
	ID: makeNotes [] _ _ :: [Int] -> [Int] -> [Beat] -> [Note]
	ID: makeNotes _ [] _ :: [Int] -> [Int] -> [Beat] -> [Note]
	ID: makeNotes _ _ [] :: [Int] -> [Int] -> [Beat] -> [Note]
	ID: makeNotes pc : pcs reg : regs bt : bts :: [Int] -> [Int] -> [Beat] -> [Note]
	ID: tail hd : tl :: [a] -> [a]
	ID: head hd : tl :: [a] -> a
	ID: reverse [] :: [a] -> [a]
	ID: reverse a : rest :: [a] -> [a]
	ID: drop 0 x :: Int -> [Note] -> [Note]
	ID: drop _ [] :: Int -> [Note] -> [Note]
	ID: drop x l : rest :: Int -> [Note] -> [Note]
	ID: take 0 _ :: Int -> [Note] -> [Note]
	ID: take _ [] :: Int -> [Note] -> [Note]
	ID: take x l : rest :: Int -> [Note] -> [Note]
	ID: track1  :: System
	ID: track2  :: System
	ID: track3  :: System
	ID: track4  :: System
	ID: print x :: a -> a
	ID: random  :: Int

java -jar ./Lib/CSV2MIDI.jar a.csv a.midi
===== Program Successfully Finished =====
===== Result Writen to a.midi =====
